core/domain/summary_services.py

```diff
@@ -415,32 +416,36 @@ def get_featured_activity_summary_dicts(language_codes):
     return featured_summary_dicts
 
 
-def get_top_rated_exploration_summary_dicts(language_codes):
+def get_top_rated_exploration_summary_dicts(
+        language_codes, number_of_explorations_to_fetch):
     """Returns a list of top rated explorations with the given language code.
 
     The return value is sorted in decreasing order of average rating.
     """
     filtered_exp_summaries = [
         exp_summary for exp_summary in
-        exp_services.get_top_rated_exploration_summaries().values()
+        exp_services.get_top_rated_exploration_summaries(
+            number_of_explorations_to_fetch).values()
         if exp_summary.language_code in language_codes and
         sum(exp_summary.ratings.values()) > 0]
 
     sorted_exp_summaries = sorted(
         filtered_exp_summaries,
         key=lambda exp_summary: exp_summary.scaled_average_rating,
-        reverse=True)[:feconf.NUMBER_OF_TOP_RATED_EXPLORATIONS]
+        reverse=True)[:number_of_explorations_to_fetch]
```

Comment:
But you've already filtered to a max limit above, right -- so this is unnecessary?